insteon_mqtt/device/HiddenDoor.py

```diff
@@ -0,0 +1,652 @@
+
+    The device will broadcast messages on the following groups:
+      group 01 = Open [One or Two Group Mode] (0x11) / Closed [One Group
+      Mode Only] (0x13)
+      group 02 = Closed [Two Group Mode Only] (0x11)
+      group 03 = Low battery (0x11) / Good battery (0x13)
+      group 04 = Heartbeat (0x11)
+
+    State changes are communicated by emitting signals.  Other classes can
+    connect to these signals to perform an action when a change is made to
+    the device (like sending MQTT messages).  Supported signals are:
+
+    - signal_on_off( Device, bool is_on ): Sent when the sensor is tripped
+      (is_on=True) or resets (is_on=False).
+
+    - signal_low_battery( Device, bool is_low ): Sent to indicate the current
+      battery state.
+
+    - signal_heartbeat( Device, True ): Sent when the device has broadcast a
+      heartbeat signal.
+    """
+    type_name = "hidden_door"
+
+    # This defines what is the minimum time between battery status requests
+    # for devices that support it.  Value is in seconds
+    # Currently set at 1 minute
+    BATTERY_TIME = 60
+
+    def __init__(self, protocol, modem, address, name=None):
+        """Constructor
+
+        Args:
+          protocol (Protocol):  The Protocol object used to communicate
+                   with the Insteon network.  This is needed to allow the
+                   device to send messages to the PLM modem.
+          modem (Modem):  The Insteon modem used to find other devices.
+          address (Address): The address of the device.
+          name (str):  Nice alias name to use for the device.
+        """
+        super().__init__(protocol, modem, address, name)
+
+        self.signal_dawn = Signal()
+        self.signal_voltage = Signal()
+        self.signal_low_voltage_threshold = Signal()
+        self.signal_heartbeat_interval = Signal()
+
+        # Insert the 'Two Groups' closed callback on group 02.  Base class
+        # already handles the other groups.
+        self.group_map.update({0x02 : self.handle_closed})
+
+        # Remote (mqtt) commands mapped to methods calls.  Add to the
+        # base class defined commands.
+        self.cmd_map.update({
+            'set_flags' : self.set_flags,
+            'set_heart_beat_interval': self.set_heart_beat_interval,
+            'set_low_battery_voltage': self.set_low_battery_voltage,
+            'get_battery_voltage' : self._get_ext_flags,
+            'get_flags' : self._get_ext_flags,
+            })
+
+        # This allows for a short timer between sending automatic battery
+        # requests.  Otherwise, a request may get queued multiple times
+        self._battery_request_time = 0
+
+    #-----------------------------------------------------------------------
+    @property
+    def battery_voltage_time(self):
+        """Returns the timestamp of the last battery voltage report from the
+        saved metadata
+        """
+        meta = self.db.get_meta('HDS')
```

Comment:
In a [comment](https://github.com/TD22057/insteon-mqtt/pull/285#issuecomment-751823375) on #285, @krkeegan indicated that he prefers a flatter metadata format.  It may be worth considering doing that for this device, since it's new code being added.  I'll leave it for you and Kevin to decide, though.

Reply:
Thanks for this.  I basically copied this out of the motion sensor that already existed.  I'll let @krkeegan tell me if he wants something different.

Reply:
Yeah, @tstabrawa is right.  I was wrong in writing that in leak and now I have to figure out a way to back that out of people's data files.

If you can just change to like 
```suggestion
        time = self.db.get_meta('HDS_battery_voltage_time')
```

Same thing on the setter, no need to wrap it in a dictionary.

I can also fix it if you like.

Reply:
It might be better if you fix this.  I'm not sure I follow completely.

Reply:
I think I have everything done other than this item.  I'll leave this one to you if that's OK.